### PR TITLE
Fix nompi builds for fluidfft v0.3.2

### PR DESCRIPTION
**Pull request by ashwinvis**

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -95,14 +95,14 @@ test:
 
 about:
   home: https://bitbucket.org/fluiddyn/fluidfft
```
Comment (paugier) [inline recipe/meta.yaml]: Better with the foss.heptapod.net address?

File: recipe/meta.yaml [hunk 2]
```diff
@@ -63,10 +63,10 @@ requirements:
     - libblas
     - libcblas
   run:
+    - python >=3.6
```
Comment (paugier) [inline recipe/meta.yaml]: I think it should just be `python` here. And add `<3.6` to `skip`.

File: recipe/meta.yaml [hunk 3]
```diff
@@ -95,14 +95,14 @@ test:
 
 about:
   home: https://bitbucket.org/fluiddyn/fluidfft
-  license: CeCILL-B
-  license_file: LICENSE.txt 
+  license: CECILL-2.1
```
Comment (paugier) [inline recipe/meta.yaml]: See for license info that makes the bot happy https://github.com/conda-forge/fluidsim-feedstock/pull/3/files#diff-f3725a55bf339595bf865fec73bda8ac99f283b0810c205442021f29c06eea9a

And the License file has to be added into the recipe directory.

**Comment by paugier** [comment]
It would be great to fix this recipe, especially now that fluiddyn, fluidsim and fluidimage recipes are in a good state.

@CyrilleBonamy could you please help on this one. You are our specialist in conda-forge recipes :slightly_smiling_face: 

Fluidfft is this only package for which all the MPI complication is needed, because it contains extensions compiled against an MPI library. Maybe we have to look more at mpi4py recipe https://github.com/conda-forge/mpi4py-feedstock ?

**Comment by paugier** [comment]
Note that the command `conda create -n tmp_env_mpi4py mpi4py` installs
```
    package                    |            build
    ---------------------------|-----------------
    mpi-1.0                    |          openmpi           4 KB  conda-forge
    mpi4py-3.0.3               |   py39hd96e10c_2         677 KB  conda-forge
    openmpi-4.0.5              |       h2c4babf_3         3.8 MB  conda-forge
    ------------------------------------------------------------
```
which is reasonable.

**Comment by CyrilleBonamy** [comment]
For nompi build, why fftw3_mpi is set to True in site.cfg?

**Comment by paugier** [comment]
See also https://github.com/conda-forge/fftw-feedstock/tree/master/recipe and the corresponding files https://anaconda.org/conda-forge/fftw/files

**Comment by paugier** [comment]
> For nompi build, why fftw3_mpi is set to True in site.cfg?

Ah you are right `fftw3_mpi` should be set to False in site.cfg for nompi builds. Where/how is it set to True ?

**Comment by CyrilleBonamy** [comment]
somewhere here  :  https://pypi.io/packages/source/f/fluidfft/fluidfft-0.3.2.tar.gz

**Comment by CyrilleBonamy** [comment]
The problem is here :  fluidfft-0.3.2/site.cfg.files/site.cfg.travisXXX (i guess)

**Comment by paugier** [comment]
I guess we should have special site.cfg files in this repository.

**Comment by ashwinvis** [comment]
I had a look and the `build.sh` and `bld.bat` is responsible for generating the site.cfg files. It looks good to me and it beats me how it fails.

**Comment by ashwinvis** [comment]
@conda-forge/fluidfft All linux and some osx builds are fixed now. osx with python 3.6 fails [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=236071&view=logs&j=9c5ef928-2cd6-52e5-dbe6-9d173a7d951b&t=20c71c51-4b27-578b-485d-06ade2de1d00&l=1374) and [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=236071&view=logs&j=15f8dc28-9f6f-5e3a-5b3c-2b169071e5be&t=63fa899e-c160-5185-75c4-c5fd30c90374&l=1326) and [here](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=236071&view=logs&j=c35410bf-a820-56fd-e231-d3cd4bead63c&t=b92b255f-9dfe-5c14-4730-460f7a671001&l=1357). The windows builds are ~~more cryptic~~ failing because pythran is not provided: so either we supply the necessary environment variable or we try enabling pythran as a build dependency.

Any suggestions?

**Comment by paugier** [comment]
Using clang, I think there should be no problem about using Pythran on Windows, but it's true that it would already be a great + to have packages for Linux, OSX and Windows so a first version with `DISABLE_PYTHRAN` for Windows would be totally fine!

**Comment by paugier** [comment]
> osx with python 3.6 fails

Easy solution: skip this build! We don't need to support this case if we have packages for OSX and Python 3.7 and 3.8 !

**Comment by ashwinvis** [comment]
@paugier This has to be fixed upstream:

```py
%SRC_DIR%>set "CXX=cl.exe" 

%SRC_DIR%>set "CC=cl.exe" 

%SRC_DIR%>set "VSINSTALLDIR=" 

%SRC_DIR%>for /F "usebackq tokens=*" %i in (`vswhere.exe -nologo -products * -version [15.0,16.0) -property installationPath`) do (set "VSINSTALLDIR=%i\" ) 

%SRC_DIR%>(set "VSINSTALLDIR=C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\" ) 

%SRC_DIR%>if not exist "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\" (for /F "usebackq tokens=*" %i in (`vswhere.exe -nologo -products * -requires Microsoft.VisualStudio.Component.VC.v141.x86.x64 -property installationPath`) do (set "VSINSTALLDIR=%i\" ) ) 

%SRC_DIR%>if not exist "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\" (set "VSINSTALLDIR=C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional\" ) 

%SRC_DIR%>if not exist "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\" (set "VSINSTALLDIR=C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\" ) 

%SRC_DIR%>if not exist "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\" (set "VSINSTALLDIR=C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools\" ) 

%SRC_DIR%>if not exist "C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\" (set "VSINSTALLDIR=C:\Program Files (x86)\Microsoft Visual Studio\2017\Enterprise\" ) 

%SRC_DIR%>IF NOT "1" == "" (
set "INCLUDE=%PREFIX%\Library\include;%PREFIX%\Library\include;"  
    File ".\setup_make.py", line 271, in <listcomp>
      command = " ".join([conf_vars[k] for k in keys])
  KeyError: 'CXX'
  Building wheel for fluidfft (PEP 517): finished with status 'error'
```

**Comment by ashwinvis** [comment]
Fixed what is possible with this version. Linux and osx are on par. Windows package only has the pure python pyfftw backend enabled. Good to go.

**Comment by paugier** [comment]
Thanks a lot @ashwinvis !

**Pull request merged**